source/common/common/logger.h

```diff
@@ -404,9 +404,11 @@ template <Id id> class Loggable {
 
 #define ENVOY_LOG_FIRST_N(LEVEL, N, ...)                                                           \
   do {                                                                                             \
-    static auto* countdown = new std::atomic<uint64_t>();                                          \
-    if (countdown->fetch_add(1) < N) {                                                             \
-      ENVOY_LOG(LEVEL, ##__VA_ARGS__);                                                             \
+    if (ENVOY_LOG_COMP_LEVEL(ENVOY_LOGGER(), LEVEL)) {                                             \
+      static auto* countdown = new std::atomic<uint64_t>();                                        \
```

Comment:
Following safety standards I'd recommend changing the type to `int64_t` (`signed` type) everywhere where arithmetic is involved, like a default counter here (`unsigned` types are allowed for bitwise operations).